index.js

```diff
@@ -23,7 +23,8 @@ module.exports = function argsert (typeConfig, ...args) {
     const observedType = Array.isArray(arg) ? 'array'
       : arg === null ? 'null'
         : arg instanceof Error ? 'error'
-          : typeof arg;
+          : arg instanceof Promise ? 'promise'
```

Comment:
I wouldn't do an `instanceof` check to match promises as that locks out Bluebird created promises. Maybe check to see if `typeof obj.then === 'function'`?

This has the benefit of also not breaking older nodes.